toolchain/check/handle_operator.cpp

```diff
@@ -314,6 +314,18 @@ auto HandleParseNode(Context& context, Parse::PrefixOperatorPartialId node_id)
     -> bool {
   auto value_id = context.node_stack().PopExpr();
   auto inner_type = ExprAsType(context, node_id, value_id);
+  auto class_type =
+      context.types().TryGetAs<SemIR::ClassType>(inner_type.type_id);
+
+  if (!class_type ||
+      context.classes().Get(class_type->class_id).inheritance_kind ==
+          SemIR::Class::InheritanceKind::Final) {
+    CARBON_DIAGNOSTIC(PartialOnFinal, Error,
+                      "`partial` applied to final type {0}", SemIR::TypeId);
+    context.emitter().Emit(LocIdForDiagnostics::TokenOnly(node_id),
```

Comment:
```suggestion
    context.emitter().Emit(node_id,
```

Drive-by -- is it fair to assume that `partial` doesn't have children, and so `TokenOnly` doesn't need to be done here? (I believe it's mainly for nodes with children, when we don't want to include the children)

Reply:
I think the `T` in `partial T` would be a child, so this'd change the diagnostic from underlining just the `partial` to underlining the type as well. Maybe that's an improvement though?

Reply:
Ahhh, yeah you're right. But yeah, maybe it should because the argument may be less obviously the type in the output.

Reply:
Yeah, that seems like an improvement - 16de6517bb723f2a1614dd1ddd00d9f922c88066